src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,205 @@
+package duke;
+
+import duke.exception.*;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+/**
+ * A Duke object which load the SoCCat,  a Personal Assistant Chatbot that helps a person to keep track of various
+ * things. The task list will be saved whenever there are changes, and will be loaded the next time the bot starts up.
+ */
+public class Duke {
+
+    private final static DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm");
+    private final Storage storage;
+    private final Ui ui;
+    private final Parser parser;
+    private TaskList tasks;
+
+    public static void main(String[] args) {
+        new Duke().start();
+    }
+
+    /**
+     * Creates a Duke object.
+     */
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage();
+        parser = new Parser();
+    }
+
+    private void start() {
+        ui.printWelcomeMessage();
+
+        try {
+            tasks = new TaskList(storage.loadFromDisk());
+            repeatUntilQuit();
+            exit();
+        } catch (DukeException ex) {
+            System.out.println(ex.getMessage());
+        }
+    }
+
+    private void exit() {
+        ui.printGoodbyeMessage();
+    }
+
+    private void repeatUntilQuit() {
+
+        while (true) {
+            String input = ui.getUserInput();
+            String[] words = parser.parseInput(input);
+            String keyword = parser.getKeyword(input);
+
+            try {
+                switch (keyword) {
+                    case "bye":
+                        return;
+                    case "list":
+                        ui.printAllTasks(tasks.getTaskList());
+                        break;
+                    case "todo":
+                        createToDos(words);
+                        break;
+                    case "deadline":
+                        createDeadlines(words);
+                        break;
+                    case "event":
+                        createEvents(words);
+                        break;
+                    case "mark":
+                        mark(words);
+                        break;
+                    case "unmark":
+                        unmark(words);
+                        break;
+                    case "delete":
+                        deleteTask(words);
+                        break;
+                    case "find":
+                        findTask(words);
+                        break;
+                    default:
+                        throw new DukeInvalidException();
+                }
```

Comment:
I noticed an extra indentation for this switch statement on each case label. Perhaps by unidenting each case label, it would better conform to the coding standards.

Reply:
I noticed a good use of abstraction and the SLAP principle here